python/ray/experimental/channel/serialization_context.py

```diff
@@ -31,6 +32,9 @@ def __init__(self):
         # The number of readers for each channel. When the number of readers
         # reaches 0, remove the data from the buffer.
         self.channel_id_to_num_readers: Dict[str, int] = {}
+        # Caching GPU futures ensures CUDA events associated with futures are propoerly
+        # destroyed instead of relying on garbage collection.
```

Comment:
Could you specify in the comment what the key of the dictionary is? And also when events are added/removed from the dictionary?

Reply:
Also, add a comment explaining what will happen if caller tries to add a future at a key that already exists.